recipes/zmqpp/all/conanfile.py

```diff
@@ -1,93 +1,99 @@
     homepage = "https://github.com/zeromq/zmqpp"
-    version = "4.2.0"
-    version_major = 4
-    license = "MPLv2"
+    license = "MPL-2.0"
     url = "https://github.com/conan-io/conan-center-index"
     description = "This C++ binding for 0mq/zmq is a 'high-level' library that hides most of the c-style interface core 0mq provides."
     topics = ("conan", "zmq", "0mq", "ZeroMQ", "message-queue", "asynchronous")
     settings = "os", "compiler", "build_type", "arch"
-    requires = "zeromq/4.3.3"
     options = {"shared": [True, False], "fPIC": [True, False]}
     default_options = {"shared": False, "fPIC": True}
-    generators = "make"
+    exports_sources = ["CMakeLists.txt"]
+    generators = "cmake", "cmake_find_package"
 
     @property
     def _source_subfolder(self):
-        return os.path.join(self.source_folder, "source_subfolder")
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
 
     def config_options(self):
         if self.settings.os == 'Windows':
             del self.options.fPIC
 
+    def requirements(self):
+        self.requires("libsodium/1.0.18")
+        self.requires("zeromq/4.3.3")
+
     def source(self):
         tools.get(**self.conan_data["sources"][self.version])
-        extracted_dir = "zmqpp-" + self.version
-        os.rename(extracted_dir, self._source_subfolder)
-
-        # zmpqq Makefile uses standard compiler variable name, useful for adding extra libs, so we rename it.
-        tools.replace_in_file("source_subfolder/Makefile", "LIBRARY_PATH = $(SRC_PATH)/$(LIBRARY_DIR)",
-                              "LIBRARY_PATH1 = $(SRC_PATH)/$(LIBRARY_DIR)\n$(info ENV==$(shell env))")
-        tools.replace_in_file("source_subfolder/Makefile", "ALL_LIBRARY_OBJECTS := $(patsubst $(SRC_PATH)/%.cpp, $(OBJECT_PATH)/%.o, $(shell find $(LIBRARY_PATH) -iname '*.cpp'))",
-                              "ALL_LIBRARY_OBJECTS := $(patsubst $(SRC_PATH)/%.cpp, $(OBJECT_PATH)/%.o, $(shell find $(LIBRARY_PATH1) -iname '*.cpp'))")
-        tools.replace_in_file("source_subfolder/Makefile", "ALL_LIBRARY_INCLUDES := $(shell find $(LIBRARY_PATH) -iname '*.hpp')",
-                              "ALL_LIBRARY_INCLUDES := $(shell find $(LIBRARY_PATH1) -iname '*.hpp')")
-        
-        # zmpqq Makefile always tries to link both library types                      
-        if self.options.shared == "True":
-            replacement = "main: $(LIBRARY_SHARED)"
+        os.rename("zmqpp-%s" % (self.version), self._source_subfolder)
+        cmakeFile = os.path.join(self._source_subfolder, "CMakeLists.txt")
+        if self.options.shared == "False": # zmqpp uses different name for static library
+            tools.replace_in_file(cmakeFile, "generate_export_header(zmqpp)", "generate_export_header(zmqpp-static)")
+            tools.replace_in_file(cmakeFile, "${CMAKE_CURRENT_BINARY_DIR}/zmqpp_export.h", "${CMAKE_CURRENT_BINARY_DIR}/zmqpp-static_export.h")
```

Comment:
Static should not require the export header 🤔 can you explain why this is required?

Reply:
export header is generated by zmqpp CMake file and it's included in zmqpp compability.hpp header. Should I try to disable in Conan recipe?

Reply:
https://cmake.org/cmake/help/latest/module/GenerateExportHeader.html... See my comment https://github.com/conan-io/conan-center-index/pull/4487#discussion_r579674341

Let's fix the other issue I found and we can come back to this one